src/main/java/org/folio/circulation/domain/LoanRepository.java

```diff
@@ -100,6 +100,30 @@ public LoanRepository(Clients clients) {
       .thenComposeAsync(r -> r.after(this::refreshLoanRepresentation));
   }
 
+  public CompletableFuture<Result<LoanAndRelatedRecords>> updateLoanAndItemInStorage(
+    LoanAndRelatedRecords relatedRecords) {
+
+    return updateLoanAndItemInStorage(relatedRecords.getLoan())
+      .thenApply(mapResult(relatedRecords::withLoan));
+  }
+
+  public CompletableFuture<Result<Loan>> updateLoanAndItemInStorage(Loan loan) {
+    if (loan == null || loan.getItem() == null) {
+      return completedFuture(succeeded(null));
+    }
+
+    //TODO: What should happen if updating the item fails?
+    return updateItem(loan.getItem())
+      .thenComposeAsync(response -> updateLoan(loan));
+  }
+
+  private CompletableFuture<Result<Response>> updateItem(Item item) {
+    if (!item.hasChanged()) {
+      return completedFuture(succeeded(null));
+    }
+    return itemRepository.updateItem(item);
+  }
```

Comment:
Why is there a need for a new way to updated the stored loan and the item?

Reply:
Because the existed method updates only a loan. In scope of this story an item should also be updated. I've moved out updateLoanAndItemInStorage method to LoanRepository.class in order to remove duplication.  Now this method is used in DeclareLostResource and RenewalResource class

Reply:
> Because the existed method updates only a loan. In scope of this story an item should also be updated. 

Why does a renewal need to change the item?

As I understand it, a renewal and an overridden renewal should likely perform very similar operations.

> I've moved out updateLoanAndItemInStorage method to LoanRepository.class in order to remove duplication. 

I'm happy with extracting this sub-process to reduce duplication. I would prefer it be it's own class, rather than the loan repository, which is intended to be specific to loans. 

Admittedly, some leaking has occurred because mod-circulation has mixed the aggregate and query models together, which eventually needs resolving.

Reply:
> Why does a renewal need to change the item?

According to the story's requirement an item status should be changed to CHECK_OUT

Reply:
> > Why does a renewal need to change the item?
> 
> According to the story's requirement an item status should be changed to CHECK_OUT

Hmm, I think this might be a matter of interpretation.

Given that a renewal is conducted on an open loan, the item should already been checked out, so it likely does not need changing again.

I think this likely needs following up with the PO.

Reply:
Do the tests fail if the item is not updated?

Reply:
Yes, they do. This case is covered in canOverrideRenewalWhenItemIsDeclaredLost test method, line # 426 in OverrideRenewByBarcodeTests class

Reply:
What status is the item in when those tests fail?

Reply:
Hmm, I think this might be a matter of interpretation.

Given that a renewal is conducted on an open loan, the item should already been checked out, so it likely does not need changing again.

I think this likely needs following up with the PO.

> 

The item has DECLARED_LOST status, after a renewal the status is changed to CHECK_OUT

Reply:
> What status is the item in when those tests fail?
The item status is DECLARED_LOST

Reply:
Ok, thank you for explaining that. The requirements now make sense to me.

 I think this coordination of updates should be done in a separate class to the repository.

Reply:
Changed